### ISSUE TITLE
(collect-updates) - doesn't consider ref's in --since flag

### ISSUE THREAD

**Issue by StarpTech** [lerna/lerna#1927]
## Expected Behavior
Should display the changed packages since the glob expression defined in [describe-ref](https://github.com/lerna/lerna/tree/c2405a1d30e053684654dc3a7e2669caca846fd1/utils/describe-ref). According to the docs `lerna ls` supports all filter flags. 
```
yarn lerna list --since <tag>
yarn lerna list --since v1.0.x
yarn lerna list --since 07ad704ca1065c494f64dc7a326883fbb68003f9
```

## Current Behavior
it will output always

```
lerna notice cli v3.10.8
lerna notice Current HEAD is already released, skipping change detection.

lerna success found 0 packages
Done in 1.01s.
```

## Possible Solution
Assign the value of `since` to the `match` option of the `execOpts` options. In that way the describe-ref utility will respect the `match` flag. 
https://github.com/lerna/lerna/blob/c2405a1d30e053684654dc3a7e2669caca846fd1/utils/collect-updates/collect-updates.js#L23

| Executable | Version |
| ---: | :--- |
| `lerna --version` | v3.10.8 |
| `npm --version`  | v1.12.1 |

| OS | Version |
| --- | --- |
| NAME | VERSION |
| Windows 10 | 1607 |

**Comment by StarpTech** [comment]
@evocateur would you accept a PR?

**Comment by evocateur** [comment]
The argument to `--since` has never claimed to be a glob, unless I'm mistaken?

**Comment by StarpTech** [comment]
Hi,  with `--since` we should be able to pass a `ref` right? If I pass a commit id or a tag it doesn't work but with the change it does. I thought that all parameters of `describe-ref` should be supported isn't?

**Comment by StarpTech** [comment]
> The argument to `--since` has never claimed to be a glob, unless I'm mistaken?

Yes, it's not about the glob but `collect-updates` does not respect the `match` feature by the `describe-ref``utility. This is very useful in our scenarios. Would you accept a PR to fix it or is `since` the wrong flag?

**Comment by evocateur** [comment]
`collect-updates` doesn't use the `match` option of `describe-ref`, and it shouldn't:

1. `--since` can be _any_ git ref, not just a tag name.
2. `options.match` is passed directly to [`git describe --match`](https://www.git-scm.com/docs/git-describe#git-describe---matchltpatterngt), which needs to be a glob.
3. `--since` isn't a glob.

That said, I think I have a fix:
```patch
diff --git a/utils/collect-updates/__tests__/collect-updates.test.js b/utils/collect-updates/__tests__/collect-updates.test.js
index 42ec603..63452e7 100644
--- a/utils/collect-updates/__tests__/collect-updates.test.js
+++ b/utils/collect-updates/__tests__/collect-updates.test.js
@@ -257,6 +257,29 @@ describe("collectUpdates()", () => {
     expect(makeDiffPredicate).toHaveBeenLastCalledWith("beefcafe", execOpts, undefined);
   });
 
+  it("does not exit early on tagged release when --since <ref> is passed", () => {
+    changedPackages.add("package-dag-1");
+
+    describeRef.sync.mockReturnValueOnce({
+      refCount: "0",
+    });
+
+    const graph = buildGraph();
+    const pkgs = graph.rawPackageList;
+    const execOpts = { cwd: "/test" };
+
+    const updates = collectUpdates(pkgs, graph, execOpts, {
+      since: "deadbeef",
+    });
+
+    expect(updates).toEqual([
+      expect.objectContaining({ name: "package-dag-1" }),
+      expect.objectContaining({ name: "package-dag-2a" }),
+      expect.objectContaining({ name: "package-dag-2b" }),
+      expect.objectContaining({ name: "package-dag-3" }),
+    ]);
+  });
+
   it("ignores changes matched by --ignore-changes", () => {
     const graph = buildGraph();
     const pkgs = graph.rawPackageList;
diff --git a/utils/collect-updates/collect-updates.js b/utils/collect-updates/collect-updates.js
index f0388e1..65aa5d6 100644
--- a/utils/collect-updates/collect-updates.js
+++ b/utils/collect-updates/collect-updates.js
@@ -24,7 +24,7 @@ function collectUpdates(filteredPackages, packageGraph, execOpts, commandOptions
     const { sha, refCount, lastTagName } = describeRef.sync(execOpts, commandOptions.includeMergedTags);
     // TODO: warn about dirty tree?
 
-    if (refCount === "0" && forced.size === 0) {
+    if (refCount === "0" && forced.size === 0 && !committish) {
       // no commits since previous release
       log.notice("", "Current HEAD is already released, skipping change detection.");
 
```
(the core of your issue is that you're running `lerna ls` _on_ the tagged release, not that the `git describe` needs matching)

**Comment by evocateur** [comment]
published in [v3.12.1](https://github.com/lerna/lerna/releases/tag/v3.12.1)

**Comment by StarpTech** [comment]
Yeah, that looks reasonable. Thank you!
